### PR TITLE
Explain how HStore columns can be queried [ci skip]

### LINKED ISSUE #22876: Psql confused by hstore attribute in 'WHERE' clause
using Model.where(model).first_or_create in Rails 5beta. Trying to create new record from hash of attributes which includes an ```hstore``` column. 

*My code:*
Attribute hashes:

```Ruby
marvel = {
  name: 'Marvel',
  description: 'This is the standard shadow template of the marvel kit',
  shadow_params: {
    led: '0',
    pb: '0',
    pb_lambda: '0' }
}
arduino_lock = {
  name: 'arduino_lock',
  description: 'This is the standard shadow template of an Arduino lock',
  shadow_params: {
    door: 'closed',
    master_code: '424242',
    valid_codes: %w(254545 123456 654321)
  }
}
iphone = {
  name: 'iPhone',
  description: 'This is the standard shadow template of an iPhone for tracking',
  shadow_params: {
    location: {
      lat: '',
      lon: '',
      timestamp: ''
    }
  }
}
```

*Then doing a batch first or create on those hashes:*

```Ruby
[marvel, arduino_lock, iphone].each do |type|
  ThingType.where(type).first_or_create
end
```
**The SQL that is generated has the ```hstore``` attribute listed as a separate table instead of a column.**

```SQL
 SELECT "things".* FROM "things" WHERE "things"."name" = $1 AND "things"."description" = $2 AND "shadow_params"."led" = $3 AND "shadow_params"."pb" = $4 AND "shadow_params"."pb_lambda" = $5
 ```

**Resulting in the following ```ActiveRecord ```Error:**

```console
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "shadow_params"
LINE 1: ...me" = $1 AND "thing_types"."description" = $2 AND "shadow_pa...
```

Full output
```
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "shadow_params"
LINE 1: ...me" = $1 AND "thing_types"."description" = $2 AND "shadow_pa...
                                                             ^
: SELECT "thing_types".* FROM "thing_types" WHERE "thing_types"."name" = $1 AND "thing_types"."description" = $2 AND "shadow_params"."led" = $3 AND "shadow_params"."pb" = $4 AND "shadow_params"."pb_lambda" = $5
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/postgresql_adapter.rb:628:in `prepare'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/postgresql_adapter.rb:628:in `prepare_statement'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/postgresql_adapter.rb:589:in `exec_cache'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/postgresql_adapter.rb:576:in `execute_and_clear'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/postgresql/database_statements.rb:160:in `exec_query'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/abstract/database_statements.rb:375:in `select_prepared'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/abstract/database_statements.rb:39:in `select_all'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/connection_adapters/abstract/query_cache.rb:70:in `select_all'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/querying.rb:39:in `find_by_sql'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/relation.rb:691:in `exec_queries'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/relation.rb:572:in `load'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/relation.rb:252:in `to_a'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.beta1/lib/active_record/relation.rb:675:in `inspect'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-5.0.0.beta1/lib/rails/commands/console.rb:65:in `start'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-5.0.0.beta1/lib/rails/commands/console_helper.rb:9:in `start'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-5.0.0.beta1/lib/rails/commands/commands_tasks.rb:78:in `console'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-5.0.0.beta1/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-5.0.0.beta1/lib/rails/command.rb:20:in `run'
	from /Users/User/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-5.0.0.beta1/lib/rails/commands.rb:19:in `<top (required)>'
	from bin/rails:4:in `require'
```

### PR DESCRIPTION
**Pull request by prathamesh-sonpatki**
- Fixes #22876.

### CODE REVIEW

**Comment by prathamesh-sonpatki** [comment]
@senny Looks like this guide is not linked from anywhere. We can add it under http://edgeguides.rubyonrails.org/active_record_querying.html or under `Digging Deeper` section.

**Comment by senny** [comment]
@prathamesh-sonpatki the guide is not linked because it's not yet in a published state. Nonetheless it's helpful to as a reference for more advanced users.

File: guides/source/active_record_postgresql.md [hunk 1]
```diff
@@ -110,6 +110,15 @@ profile.settings = {"color" => "yellow", "resolution" => "1280x1024"}
 profile.save!
 ```
 
+#### Querying HStore columns
```
Comment (senny) [inline guides/source/active_record_postgresql.md]: I don't think we need a separate heading. Simply expand the example given above. Also make sure to add a link to available functions and operators at the beginning of the `hstore` section.

**Comment by prathamesh-sonpatki** [comment]
@senny Updated

**Comment by senny** [comment]
@prathamesh-sonpatki thank you :yellow_heart:

**Pull request merged**